docs/spec/metrics/metric.json

```diff
@@ -0,0 +1,38 @@
+{
+    "$schema": "http://json-schema.org/draft-04/schema#",
+    "$id": "docs/spec/metrics/metric.json",
+    "type": "object",
+    "description": "Metric data captured by an APM agent",
+    "properties": {
+        "samples": {
+            "type": "object",
+            "description": "Sampled application metrics collected from the agent",
+            "regexProperties": true,
+            "patternProperties": {
+                "^[^*\"]*$": {
+                    "$ref": "sample.json",
+                    "maxLength": 1024
+                }
+            },
+            "additionalProperties": false
+        },
+        "tags": {
+            "type": ["object", "null"],
+            "description": "A flat mapping of user-defined tags with string values.",
+            "regexProperties": true,
+            "patternProperties": {
+                "^[^*\"]*$": {
+                    "type": "string",
+                    "maxLength": 1024
+                }
+            },
+            "additionalProperties": false
+        },
+        "timestamp": {
+            "type": "string",
+            "format": "date-time",
+            "pattern": "Z$",
+            "description": "Recorded time of the metric, UTC based and formatted as YYYY-MM-DDTHH:mm:ss.sssZ"
+        }
+    }
```

Comment:
The comment above regarding adding `null` as a possible type for all properties should also apply to all the other properties.